src/Npgsql/NpgsqlParameterCollection.cs

```diff
@@ -325,24 +321,14 @@ public override int IndexOf(string parameterName)
         public override void RemoveAt(int index)
         {
             if (_internalList.Count - 1 < index)
-                throw new IndexOutOfRangeException();
+                throw new ArgumentOutOfRangeException(nameof(index));
+
             Remove(_internalList[index]);
         }
 
         /// <inheritdoc />
         public override void Insert(int index, object value)
-        {
-            if (value == null)
-                throw new ArgumentNullException(nameof(value));
-            if (!(value is NpgsqlParameter param))
-                throw new InvalidCastException($"{nameof(value)} must be an NpgsqlParameter");
-            if (param.Collection != null)
```

Comment:
Isn't this a breaking change? Any reason to force users to remove the parameter from the previous list before adding to the new one? IIRC this is how SqlClient works, but it doesn't mean we have to do the same...

Reply:
Sorry, got confused - this was already this way.